packages/web3-eth-contract/src/types.ts

```diff
@@ -88,26 +180,308 @@ export type PayableTxOptions = PayableCallOptions;
+	 */
 	estimateGas<ReturnFormat extends DataFormat = typeof DEFAULT_RETURN_FORMAT>(
 		options?: NonPayableCallOptions,
 		returnFormat?: ReturnFormat,
 	): Promise<FormatType<Numbers, ReturnFormat>>;
+
+	/**
+	 * Encodes the ABI for this method. The resulting hex string is 32-bit function signature hash plus the passed parameters in Solidity tightly packed format.
+	 * This can be used to send a transaction, call a method, or pass it into another smart contract’s method as arguments.
+	 * Set the data field on `web3.eth.sendTransaction` options as the encodeABI() result and it is the same as calling the contract method with `contract.myMethod.send()`.
+	 *
+	 * Some use cases for encodeABI() include: preparing a smart contract transaction for a multisignature wallet,
+	 * working with offline wallets and cold storage and creating transaction payload for complex smart contract proxy calls.
+	 *
+	 * @returns - The encoded ABI byte code to send via a transaction or call.
+	 */
 	encodeABI(): string;
 }
 
 export interface PayableMethodObject<Inputs = unknown[], Outputs = unknown[]> {
 	arguments: Inputs;
+	/**
+	 * Will call a method and execute its smart contract method in the EVM without sending any transaction. Note calling cannot alter the smart contract state.
+	 *
+	 * ```ts
+	 * // using the promise
+	 * const result = await myContract.methods.myMethod(123).call({from: '0xde0B295669a9FD93d5F28D9Ec85E40f4cb697BAe'});
+	 *
+	 * // MULTI-ARGUMENT RETURN:
+	 * // Solidity
+	 * contract MyContract {
+	 *   function myFunction() returns(uint256 myNumber, string myString) {
+	 *       return (23456, "Hello!%");
+	 *   }
+	 * }
+	 *
+	 * // web3.js
+	 * var MyContract = new web3.eth.Contract(abi, address);
+	 * const result = MyContract.methods.myFunction().call()
+	 * console.log(result)
+	 * > Result {
+	 *   myNumber: '23456',
+	 *   myString: 'Hello!%',
+	 *   0: '23456', // these are here as fallbacks if the name is not know or given
+	 *   1: 'Hello!%'
+	 * }
+	 *
+	 *
+	 * // SINGLE-ARGUMENT RETURN:
+	 * // Solidity
+	 * contract MyContract {
+	 *   function myFunction() returns(string myString) {
+	 *       return "Hello!%";
+	 *   }
+	 * }
+	 *
+	 * // web3.js
+	 * const MyContract = new web3.eth.Contract(abi, address);
+	 * const result = await MyContract.methods.myFunction().call();
+	 * console.log(result);
+	 * > "Hello!%"
+	 * ```
+	 *
+	 * @param tx - The options used for calling.
+	 * @param block - If you pass this parameter it will not use the default block set with contract.defaultBlock. Pre-defined block numbers as `earliest`, `latest`, and `pending` can also be used. Useful for requesting data from or replaying transactions in past blocks.
+	 * @returns - The return value(s) of the smart contract method. If it returns a single value, it’s returned as is. If it has multiple return values they are returned as an object with properties and indices.
+	 */
 	call(tx?: PayableCallOptions, block?: BlockNumberOrTag): Promise<Outputs>;
```

Comment:
```suggestion
	call(tx?: PayableCallOptions | NonPayableCallOptions, block?: BlockNumberOrTag): Promise<Outputs>;
```

and you can remove line # 229

Reply:
These types are initialised based on the ABI, so one method can either have one call options at same time. That's why can't use `|` here.

Reply:
Initialized means these types are created with some ABI? If so could you point to ABI source.